am-ui/webcomponents/search-webcomponent/src/app/search-form/search-form.component.spec.ts

```diff
@@ -49,33 +46,151 @@ describe('SearchFormComponent', () => {
+      options: Array(0),
+      order: 13,
+      request: true,
+      required: false,
+      searchable: false,
+      type: 2,
+    } as FormFieldBase<string>;
+
+
+    it('should save data into payload if component received data from field fromSize', () => {
+      component.form = formBuilder.group({
+        fromSize: "1111",
+        requestOrOffer: "OFFER",
+        subCategoryDropdown: 1
+      })
+
+      component.selectedCategorySearchFields = [formFieldBase1, formFieldBase2, formFieldBase3];
+      component.currentCategoryId = 1;
+      component.currentSubCategoryId = 2;
+
+      component.ngOnInit();
+      component.onSubmit();
+      expect(component.payLoad.category_id).toEqual(1);
+      expect(component.payLoad.subcategory_id).toEqual(2);
+      expect(component.payLoad.search_values?.length).toEqual(1);
+      if (component.payLoad.search_values !== undefined) {
+        expect(component.payLoad.search_values[0].field_id).toEqual(25);
+        expect(component.payLoad.search_values[0].value).toEqual("1111");
+      }
     });
 
+
+    it('should save data into payload if component received data from field radius', () => {
+      component.form = formBuilder.group({
+        radius: "2222",
+        requestOrOffer: "OFFER",
+        subCategoryDropdown: 1
+      })
+
+      component.selectedCategorySearchFields = [formFieldBase1, formFieldBase2, formFieldBase3];
+      component.currentCategoryId = 1;
+      component.currentSubCategoryId = 2;
+
+      component.ngOnInit();
+      component.onSubmit();
+      expect(component.payLoad.category_id).toEqual(1);
+      expect(component.payLoad.subcategory_id).toEqual(2);
+      expect(component.payLoad.search_values?.length).toEqual(1);
+      if (component.payLoad.search_values !== undefined) {
+        expect(component.payLoad.search_values[0].field_id).toEqual(34);
+        expect(component.payLoad.search_values[0].value).toEqual("2222");
+      }
+
+      expect(component.payLoad.search_values?.length).toEqual(1);
+    });
+
+    it('should save data into payload if component received data from field toPrice', () => {
+      component.form = formBuilder.group({
+        toPrice: "3333",
+        requestOrOffer: "OFFER",
+        subCategoryDropdown: 1
+      })
+
+      component.selectedCategorySearchFields = [formFieldBase1, formFieldBase2, formFieldBase3];
+      component.currentCategoryId = 1;
+      component.currentSubCategoryId = 2;
+
+      component.ngOnInit();
+      component.onSubmit();
+      expect(component.payLoad.category_id).toEqual(1);
+      expect(component.payLoad.subcategory_id).toEqual(2);
+      expect(component.payLoad.search_values?.length).toEqual(1);
+      if (component.payLoad.search_values !== undefined) {
+        expect(component.payLoad.search_values[0].field_id).toEqual(24);
+        expect(component.payLoad.search_values[0].value).toEqual("3333");
+      }
+    });
+
+    // it('should be created and not show form', () => {
```

Comment:
You could remove line 174 to 176. I don't think there is any point in adding a commented test.

Reply:
This commented text was not written by me. Should I delete it?

Reply:
Sorry, I thought it was. I would say if the test case is not easily fixable then it should be deleted. But it's up to you.